libcxx/test/std/iterators/iterator.primitives/iterator.traits/cxx20_iterator_traits.compile.pass.cpp

```diff
@@ -152,8 +147,7 @@ static_assert(testIOIterator<std::insert_iterator<std::vector<int>>, std::output
 static_assert(testConst<std::istream_iterator<int, char>, std::input_iterator_tag, int>());
 
 #if !defined(TEST_HAS_NO_LOCALIZATION)
-// libc++-specific since pointer type is unspecified:
-LIBCPP_STATIC_ASSERT(test<std::istreambuf_iterator<char>, std::input_iterator_tag, char, long long, char, char*>());
+static_assert(test<std::istreambuf_iterator<char>, std::input_iterator_tag, char, std::streamoff, char, char*>());
```

Comment:
Ok. Thanks for pointing it out. I thought the `LIBCPP_STATIC_ASSERT` because of `long long`. I will put it back there.

Reply:
I suggest we do this instead:

```
// We use std::istreambuf_iterator<char>::pointer because it's unspecified, it doesn't have to be char*
static_assert(test<std::istreambuf_iterator<char>, std::input_iterator_tag, char, std::streamoff, char, std::istreambuf_iterator<char>::pointer>());
```

We don't need to pin down `char*` here. We already have a test that ensures that libc++ is using `char*`: https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/iterators/stream.iterators/istreambuf.iterator/types.pass.cpp#L49

Reply:
That way, the test should be correct everywhere, we get coverage on all implementations and we don't lose any coverage for libc++.

Reply:
Thanks for noticing this BTW @cpplearner